source/tests/integration/ni_fake_non_ivi_service_tests_endtoend.cpp

```diff
@@ -0,0 +1,80 @@
+#include <gtest/gtest.h>
+#include <nifake_non_ivi/nifake_non_ivi_mock_library.h>
+#include <nifake_non_ivi/nifake_non_ivi_service.h>
+#include <server/async_processing_loop.h>
+
+#include <chrono>
+#include <memory>
+namespace ni {
+namespace tests {
+namespace integration {
+
+using namespace nifake_non_ivi_grpc;
+
+// Test NiFakeNonIviService using a client stub.
```

Comment:
The async implementation is so dependent on grpc infrastructure (completion queues, async writers, etc) that it doesn't make sense to try to test it in the "normal" fake unit test style.

Instead, this is more like the system tests, but it's using a fake service implementation.

The other advantage is that this doesn't depend too much on internals. That's especially useful right now because I'm not sure this is a final design.